### PR TITLE
nav restructure

### PR DESCRIPTION
**Pull request by shorsher**
- initialization handling
 - namespace is now defined in the route

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -21,27 +21,130 @@ import {
...
+  const [namespaces, setNamespaces] = useState<INamespace[]>([]);
+  const [selectedNamespace, setSelectedNamespace] = useState<string>('');
+  const [dataView, setDataView] = useState<DataView>('timeline');
+  const ws = useRef<ReconnectingWebsocket | null>(null);
+  const [identity, setIdentity] = useState('');
+  const [lastEvent, setLastEvent] = useState<any>();
+  const [message, setMessage] = useState('');
+  const [messageType, setMessageType] = useState<SnackbarMessageType>('error');
+  const [orgName, setOrgName] = useState('');
+  const [createdFilter, setCreatedFilter] =
```
Comment (eberger727) [inline src/App.tsx:61]: Add ability to change time range in the dropdown

File: src/App.tsx [hunk 2]
```diff
@@ -21,27 +21,130 @@ import {
...
+    Promise.all([
+      fetchWithCredentials('/api/v1/namespaces'),
+      fetchWithCredentials('/api/v1/status'),
+    ])
+      .then(async ([namespaceResponse, statusResponse]) => {
+        if (namespaceResponse.ok && statusResponse.ok) {
+          const status: IStatus = await statusResponse.json();
+          setIdentity(status.org.identity);
+          setOrgName(status.org.name);
+          setSelectedNamespace(status.defaults.namespace);
```
Comment (eberger727) [inline src/App.tsx:75]: Namespace picker should also be implemented

File: src/components/Navigation/ActivityNav.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      <ListItemButton onClick={() => setActivityOpen(!activityOpen)}>
+        <ListItemIcon>{<ChartBoxOutline />}</ListItemIcon>
+        <ListItemText>
+          <Typography>{t('activity')}</Typography>
+        </ListItemText>
+        {activityOpen ? <ExpandLess /> : <ExpandMore />}
+      </ListItemButton>
+
+      <Collapse in={activityOpen} unmountOnExit>
+        <NavItem
```
Comment (eberger727) [inline src/components/Navigation/ActivityNav.tsx:57]: Can put the main data into an array and `.map()` over it to reduce duplication

File: src/components/Navigation/NavItem.tsx [hunk 4]
```diff
@@ -14,23 +14,27 @@
...
-    /* default value */
-  },
-  namespaces: [],
-  setSelectedNamespace: () => {
-    /* default value */
-  },
-});
+export const NavItem = ({ name, action, icon, itemIsActive }: Props) => {
+  return (
+    <ListItemButton onClick={action}>
```
Comment (eberger727) [inline src/components/Navigation/NavItem.tsx:33]: Add styling for active items

File: src/pages/Activity/views/Dashboard.tsx [hunk 5]
```diff
@@ -38,12 +46,23 @@ export const ActivityDashboard: () => JSX.Element = () => {
...
+  }, []);
+
   return (
     <>
-      <Header title={'Timeline'} subtitle={'Activity'}></Header>
+      <Header title={t('timeline')} subtitle={t('activity')}></Header>
       <Grid container px={DEFAULT_PADDING}>
         <Grid container item wrap="nowrap" direction="column">
-          <ChartHeader
+          {/* <ChartHeader
```
Comment (eberger727) [inline src/pages/Activity/views/Dashboard.tsx:65]: Uncomment

**Review by dechdev — COMMENTED** [review]
Added some notes. I can take some of the items

**Pull request merged**